### PR TITLE
Flekschas/add contributing md

### PR DESCRIPTION
**Pull request by flekschas**
## Description

> What was changed in this pull request?

I added `CONTRIBUTING.md`

Based on:
- https://mozillascience.github.io/working-open-workshop/contributing/
- https://github.com/atom/atom/blob/master/CONTRIBUTING.md
- https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md

> Why is it necessary?

Good practice.

## Checklist

- ~[ ] Unit tests added or updated~
- [x] Documentation added or updated
- ~[ ] Example added or updated~
- ~[ ] Update schema.json if there are changes to the viewconf JSON structure format~
- ~[ ] Screenshot for visual changes (e.g. new tracks)~
- ~[ ] Updated CHANGELOG.md~

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+## Contribute Code 👩‍💻👨‍💻
+
+If you have improvements to HiGlass, send us your pull requests! If you're wondering what a _pull request_ is please checkout [GitHub's how to](https://help.github.com/articles/using-pull-requests/).
+
+HiGlass core members will be assigned to review your pull requests. Once the pull requests are approved and pass continuous integration checks, we will merge your pull request add it to the next scheduled release.
+
+### Pull Request Checklist ✅
+
+- Ensure that the title of incomplete pull requests starts with `[WIP]` (work in progress)
+- Ensure that your pull request answers all questions provided in the [pull request template](PULL_REQUEST_TEMPLATE.md).
```
Comment (pkerpedjiev) [inline CONTRIBUTING.md]:
```suggestion
- Ensure that your pull request answers the relevant questions in the [pull request template](PULL_REQUEST_TEMPLATE.md).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+- **Docs**: https://docs.higlass.io
+- **Development Docs**: https://docs-develop.higlass.io (related to not-yet-released code)
+- **Bugs / Issue Tracker**: https://github.com/higlass/higlass/issues
+- **Community**: https://higlass.slack.com [Join Now!](http://bit.ly/higlass-slack)
+- **News / Updates**: https://twitter.com/higlass_io
+
+## Contribute Code 👩‍💻👨‍💻
+
+If you have improvements to HiGlass, send us your pull requests! If you're wondering what a _pull request_ is please checkout [GitHub's how to](https://help.github.com/articles/using-pull-requests/).
```
Comment (pkerpedjiev) [inline CONTRIBUTING.md]:
```suggestion
If you have improvements to HiGlass, send us your pull requests! If you're wondering what a _pull request (PR)_ is please checkout [GitHub's how to](https://help.github.com/articles/using-pull-requests/).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Contributing to HiGlass
+
+🎉 Hooray, it's marvelous to see you here! 🥳 Thanks for considering to contribute to HiGlass. 👍
```
Comment (pkerpedjiev) [inline CONTRIBUTING.md]: Odd unicode character?

image

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+- Ensure that the title of incomplete pull requests starts with `[WIP]` (work in progress)
+- Ensure that your pull request answers all questions provided in the [pull request template](PULL_REQUEST_TEMPLATE.md).
+- Ensure that your pull requests successfully passes all continuous integration tests on [TravisCI](https://travis-ci.org/higlass).
+
+## Coding Style 🎨
+
+_Note, continuous integration will fail if your code is not compatible with our coding style._
+
+- **JavaScript coding style**: We are following a combination of [ESLint's recommendation](https://eslint.org/docs/rules/), [Airbnb's coding style](https://github.com/airbnb/javascript), [React's recommendation](https://github.com/yannickcr/eslint-plugin-react), and [customizations](https://github.com/higlass/higlass/blob/develop/.eslintrc#L30). The easiest way to ensure you're following our coding style is to automatically run `npm run lint`.
```
Comment (pkerpedjiev) [inline CONTRIBUTING.md]:
```suggestion
- **JavaScript coding style**: We are follow a combination of [ESLint's recommendation](https://eslint.org/docs/rules/), [Airbnb's coding style](https://github.com/airbnb/javascript), [React's recommendation](https://github.com/yannickcr/eslint-plugin-react), and [customizations](https://github.com/higlass/higlass/blob/develop/.eslintrc#L30). The easiest way to ensure you're following our coding style is to automatically run `npm run lint`.
```

**Comment by pkerpedjiev** [comment]
Oh, how about adding a `CONTRIBUTORS.md` file where every contributor could list their name (if they want).

File: CONTRIBUTING.md (hunk 4 above)
Comment (flekschas) [reply CONTRIBUTING.md]: `We are follow a combination of`? Do you mean `we are following` or `we follow`?

File: CONTRIBUTING.md (hunk 3 above)
Comment (flekschas) [reply CONTRIBUTING.md]: Screen Shot 2019-07-23 at 12 23 05 AM

I can remove the emoji :)

**Comment by flekschas** [comment]
> Oh, how about adding a CONTRIBUTORS.md file where every contributor could list their name (if they want)?

I am not of fan of this file as it's always outdated. GitHub is already providing the [contributors page](https://github.com/higlass/higlass/graphs/contributors), what would `CONTRIBUTORS.md` add to that?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- Ensure that your pull request answers the relevant questions in the [pull request template](PULL_REQUEST_TEMPLATE.md).
+- Ensure that your pull requests successfully passes all continuous integration tests on [TravisCI](https://travis-ci.org/higlass).
+
+## Coding Style 🎨
+
+_Note, continuous integration will fail if your code is not compatible with our coding style._
+
+- **JavaScript coding style**: We are following a combination of [ESLint's recommendation](https://eslint.org/docs/rules/), [Airbnb's coding style](https://github.com/airbnb/javascript), [React's recommendation](https://github.com/yannickcr/eslint-plugin-react), and [customizations](https://github.com/higlass/higlass/blob/develop/.eslintrc#L30). The easiest way to ensure you're following our coding style is to automatically run `npm run lint`.
+
+- **Python coding style**: We are following [PEP 8](https://www.python.org/dev/peps/pep-0008/) and [PEP 257](https://www.python.org/dev/peps/pep-0257/)
```
Comment (nvictus) [inline CONTRIBUTING.md]:
```suggestion
- **Python coding style**: We are following [PEP 8](https://www.python.org/dev/peps/pep-0008/) and the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) convention for [docstrings](https://www.python.org/dev/peps/pep-0257/).
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+- **Docs**: https://docs.higlass.io
+- **Development Docs**: https://docs-develop.higlass.io (related to not-yet-released code)
+- **Bugs / Issue Tracker**: https://github.com/higlass/higlass/issues
+- **Community**: https://higlass.slack.com [Join Now!](http://bit.ly/higlass-slack)
+- **News / Updates**: https://twitter.com/higlass_io
+
+## Contribute Code 👩‍💻👨‍💻
+
+If you have improvements to HiGlass, send us your pull requests! If you're wondering what a _pull request (PR)_ is please check out [GitHub's how to](https://help.github.com/articles/using-pull-requests/).
```
Comment (nvictus) [inline CONTRIBUTING.md]:
```suggestion
If you have improvements to HiGlass, send us your pull requests! If you're wondering what a _pull request (PR)_ is, please check out [GitHub's how to](https://help.github.com/articles/using-pull-requests/).
```

**Pull request merged**